### PR TITLE
Refactor User#prevNames and User#ips to arrays

### PR DESCRIPTION
**Pull request by AnnikaCodes**

### CODE REVIEW

File: server/chat-plugins/thing-of-the-day.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ class OtdHandler {
 		const nomObj = {
 			nomination: nomination,
 			name: user.name,
-			userids: Object.assign(obj, user.prevNames),
+			userids: [...user.previousIDs],
```
Comment (Zarel) [inline server/chat-plugins/thing-of-the-day.ts:159]: iirc this is still slower than `user.previousIDs.slice()` so we might want to just use that.

Comment (Zarel) [reply server/chat-plugins/thing-of-the-day.ts:159]:
```suggestion
			userids: user.previousIDs.slice(),
```

File: server/chat-plugins/trivia.ts [hunk 2]
```diff
@@ -386,29 +386,26 @@ class Trivia extends Rooms.RoomGame {
...
-
-				const tarIps = Object.keys(tarUser.ips);
-				const ipMatch = tarIps.some(ip => (ip in user.ips));
-				if (ipMatch) return this.room.tr('You have already signed up for this game.');
+			const targetUser = Users.get(id);
+			if (targetUser) {
+				const isSameUser = (
+					targetUser.previousIDs.includes(user.id) ||
+					targetUser.previousIDs.some(tarId => user.previousIDs.includes(tarId)) ||
+					Object.keys(targetUser.ips).some(ip => (ip in user.ips))
```
Comment (Zarel) [inline server/chat-plugins/trivia.ts:406]: Should `.ips` also be an array?

**Comment by Zarel** [comment]
Looks good other than those points!

File: server/chat-plugins/trivia.ts (hunk 2 above)
Comment (AnnikaCodes) [reply server/chat-plugins/trivia.ts:406]: `User#ips` also tracks the number of connections on each IP, which _seems_ useful, but isn't actually used anywhere in public or private code. I think making it an array should be fine?

**Pull request merged**